Submitted By: Ken Moffat <ken at linuxfromscratch dot org>
Date: 2017-06-05
Initial Package Version: 20170524
Upstream Status: Unknown
Origin: http://tug.org/pipermail/tex-live/2017-June/040192.html
Description: From Dr. Werner Fink of SuSe, fixes compilation with gcc-7.1

diff -Naur texlive-20170524-source.orig/texk/web2c/luatexdir/luaffi/ctype.c texlive-20170524-source/texk/web2c/luatexdir/luaffi/ctype.c
--- texlive-20170524-source.orig/texk/web2c/luatexdir/luaffi/ctype.c	2017-02-10 01:03:59.000000000 +0000
+++ texlive-20170524-source/texk/web2c/luatexdir/luaffi/ctype.c	2017-06-05 02:32:34.548531839 +0100
@@ -245,6 +245,10 @@
 
     lua_pop(L, 1); /* mt */
     cd = (struct cdata*) lua_touserdata(L, idx);
+    if (!cd) {
+        lua_pushnil(L);
+        return NULL;
+    }
     *ct = cd->type;
     lua_getuservalue(L, idx);
 
diff -Naur texlive-20170524-source.orig/texk/web2c/luatexdir/luaffi/ffi.h texlive-20170524-source/texk/web2c/luatexdir/luaffi/ffi.h
--- texlive-20170524-source.orig/texk/web2c/luatexdir/luaffi/ffi.h	2017-03-11 01:04:06.000000000 +0000
+++ texlive-20170524-source/texk/web2c/luatexdir/luaffi/ffi.h	2017-06-05 02:32:34.548531839 +0100
@@ -370,7 +370,7 @@
 #endif
 struct cdata {
     const struct ctype type
-#ifdef __GNUC__
+#if 0 /* def __GNUC__ */
       __attribute__ ((aligned(16)))
 #endif
       ;
